### PR TITLE
Improve world seeding and config handling

### PR DESCRIPTION
**Pull request by rk1a**
* distinguish between `base_seed` and `world_seed` to allow for RNG to generate world seeds from a  base seed
* add utils for minetest config reading/writing
* allow updating of minetest config entries via provided config file or dict
* fix a bug related to cleaning up world dirs 
* fix gymnasium implementation

This PR is a Ready for Review.

### CODE REVIEW

File: minetester/scripts/test_loop_parallel.py [hunk 1]
```diff
@@ -56,16 +57,16 @@ def _init():
     xserver = start_xserver(x_display)
 
     # Start loop
-    render = True
+    render = False
     obs = venv.reset()
     done = [False] * num_envs
     step = 0
     while step < max_steps:
         print(f"Elapsed steps: {venv.get_attr('_elapsed_steps')}")
```
Comment (AI-WAIFU) [inline minetester/scripts/test_loop_parallel.py:67]: I'm getting an error on this line when running the parallel test loop It's telling me get_attr doesn't exist for the AsyncVectorEnv

**Comment by AI-WAIFU** [comment]
Can you fix the merge conflicts with the latest develop?

File: minetester/scripts/test_loop_parallel.py (hunk 1 above)
Comment (rk1a) [reply minetester/scripts/test_loop_parallel.py:67]: requires gym>=0.23.1

**Review by AI-WAIFU — APPROVED** [review]
lgtm, works when tested

**Pull request merged**